--- a/cplusplus/main.cpp	2023-12-22 23:26:27.491632655 -0600
+++ b/cplusplus/main.cpp	2024-01-03 16:38:26.480651474 -0600
@@ -86,6 +86,10 @@
 
 int main(int argc, char *argv[]) {
 
+    qputenv("MAGICK_CODER_MODULE_PATH", "/app/lib/ImageMagick-7.1.1/modules-Q16HDRI/coders");
+    qputenv("MAGICK_FILTER_MODULE_PATH", "/app/lib/ImageMagick-7.1.1/modules-Q16HDRI/filters");
+    qputenv("MAGICK_CONFIGURE_PATH", "/app/lib/ImageMagick-7.1.1/config-Q16HDRI");
+
 #ifdef Q_OS_WIN
     QFileInfo f(argv[0]);
     qputenv("PATH", QString("%1;%2").arg(qgetenv("PATH"),f.absolutePath().replace("/", "\\")).toLocal8Bit());
