--- a/qml/image/PQImageNormal.qml
+++ b/qml/image/PQImageNormal.qml
@@ -196,11 +196,13 @@ Image {
                         src = PQCScriptsImages.extractMotionPhoto(deleg.imageSource)
 
                     if(src != "") {
+                        videoloader.active = false
                         // earlier versions of Qt6 seem to struggle if only one slash is used
                         if(PQCScriptsConfig.isQtAtLeast6_5())
                             videoloader.mediaSrc = "file:/" + src
                         else
                             videoloader.mediaSrc = "file://" + src
+                        videoloader.active = true
                         PQCNotify.hasPhotoSphere = false
                         return
                     }
@@ -230,10 +232,6 @@ Image {
 
         active: false
         property string mediaSrc: ""
-        onMediaSrcChanged: {
-            if(mediaSrc != "")
-                active = true
-        }
 
         asynchronous: true
         sourceComponent: motionphoto
