diff --git a/qml/image/PQImageNormal.qml b/qml/image/PQImageNormal.qml
index 5ff5249a..7412c3f3 100644
--- a/qml/image/PQImageNormal.qml
+++ b/qml/image/PQImageNormal.qml
@@ -196,7 +196,13 @@ Image {
                         src = PQCScriptsImages.extractMotionPhoto(deleg.imageSource)
 
                     if(src != "") {
-                        videoloader.mediaSrc = "file://" + src
+                        videoloader.active = false
+                        // earlier versions of Qt6 seem to struggle if only one slash is used
+                        if(PQCScriptsConfig.isQtAtLeast6_5())
+                            videoloader.mediaSrc = "file:/" + src
+                        else
+                            videoloader.mediaSrc = "file://" + src
+                        videoloader.active = true
                         PQCNotify.hasPhotoSphere = false
                         return
                     }
@@ -226,10 +232,6 @@ Image {
 
         active: false
         property string mediaSrc: ""
-        onMediaSrcChanged: {
-            if(mediaSrc != "")
-                active = true
-        }
 
         asynchronous: true
         sourceComponent: motionphoto
diff --git a/qml/image/PQVideoQt.qml b/qml/image/PQVideoQt.qml
index b15e3834..d311e5f3 100644
--- a/qml/image/PQVideoQt.qml
+++ b/qml/image/PQVideoQt.qml
@@ -24,6 +24,7 @@ import QtQuick
 import QtMultimedia
 
 import PQCScriptsFilesPaths
+import PQCScriptsConfig
 
 Item {
 
@@ -36,7 +37,8 @@ Item {
 
         id: video
 
-        source: "file://" + deleg.imageSource
+        // earlier versions of Qt6 seem to struggle if only one slash is used
+        source: (PQCScriptsConfig.isQtAtLeast6_5() ? "file:/" : "file://") + deleg.imageSource
 
         volume: PQCSettings.filetypesVideoVolume
 
@@ -56,7 +58,13 @@ Item {
 
         onPlaybackStateChanged: {
             if(playbackState === MediaPlayer.StoppedState) {
-                video.source = "file://" + deleg.imageSource
+
+                // earlier versions of Qt6 seem to struggle if only one slash is used
+                if(PQCScriptsConfig.isQtAtLeast6_5())
+                    video.source = "file:/" + deleg.imageSource
+                else
+                    video.source = "file://" + deleg.imageSource
+
                 if(PQCSettings.filetypesVideoLoop) {
                     video.play()
                 } else {
